### PR TITLE
Overhaul language interface

### PR DESCRIPTION
**Pull request by tsaglam**
The goal of this PR is to clean up the language interface:

- Remove the `usesIndex()` method as it just returns `!supportsColumns()`
- Introduce default methods for `supportsColumns()` and `isPreformatted()`
- Makes the JDoc comments of the language interface a bit more precise
- Adapts all existing frontends accordingly

### CODE REVIEW

**Review by robinmaisch — APPROVED** [review]
Looks good to me!

File: jplag.frontend-utils/src/main/java/de/jplag/Language.java [hunk 1]
```diff
@@ -41,22 +41,23 @@ public interface Language {
...
 
     /**
-     * Number of defined tokens in the scanner of the language.
+     * Determines whether a fixed-width font should be used to display that language.
      */
-    int numberOfTokens();
+    default boolean isPreformatted() {
+        return true;
+    }
 }
```
Comment (JanWittler) [inline jplag.frontend-utils/src/main/java/de/jplag/Language.java:63]: This field is never used, maybe we just drop it too?

Comment (JanWittler) [reply jplag.frontend-utils/src/main/java/de/jplag/Language.java:63]: No, the `isPreformatted`

Comment (tsaglam) [reply jplag.frontend-utils/src/main/java/de/jplag/Language.java:63]: This should be used by the report viewer in future versions. Currently, the viewer always uses a fixed-width font.

**Review by JanWittler — APPROVED** [review]

**Pull request merged**